src/exception.jl

```diff
@@ -44,6 +49,28 @@ function show(io::IO, e::PyError)
     end
 end
 
+function pyerr_construct(msg::AbstractString)
+        ptype, pvalue, ptraceback = Ref{PyPtr}(), Ref{PyPtr}(), Ref{PyPtr}()
+        # equivalent of passing C pointers &exc[1], &exc[2], &exc[3]:
+        ccall((@pysym :PyErr_Fetch), Cvoid, (Ref{PyPtr},Ref{PyPtr},Ref{PyPtr}), ptype, pvalue, ptraceback)
+        ccall((@pysym :PyErr_NormalizeException), Cvoid, (Ref{PyPtr},Ref{PyPtr},Ref{PyPtr}), ptype, pvalue, ptraceback)
+        pyerr_construct(msg, PyObject(ptype[]), PyObject(pvalue[]), PyObject(ptraceback[]))
+end
+
```

Comment:
Since this mirrors the `PyError` constructor, a typical convention would be to call it `pyerror`

Reply:
`pyerr_construct` can return either a `PyError` or a `PyJlError`. So I thought that being more explicit that this might return different types was a good thing. I'll change it to pyerror if you think that's better.

Reply:
That's why it's a lower-case constructor.

You see this a lot in `Base`.  e.g. `Base.Iterators.reverse` returns either a reversed object or an `Iterators.Reverse` object.